core/src/main/scala/org/apache/spark/ui/ToolTips.scala

```diff
@@ -62,6 +62,13 @@ private[spark] object ToolTips {
     """Time that the executor spent paused for Java garbage collection while the task was
        running."""
 
+  val PEAK_EXECUTION_MEMORY =
+    """Execution memory refers to the memory used by internal data structures created during
+       shuffles, aggregations and joins when Tungsten is enabled. The value of this accumulator
+       should be approximately the sum of the peak sizes across all such data structures created
+       in this task. For SQL jobs, this only tracks all unsafe operators, broadcast joins, and
+       external sort."""
```

Comment:
There is not a great name for "aggregation + shuffle + join". Elsewhere in Spark we actually incorrectly use "spark.shuffle.memoryFraction" for non-shuffle operations, simply because there is not a more general category. I'm not a die-hard fan of "execution memory" either, but it's the best I can come up with.